java/number-formatting.md

```diff
@@ -0,0 +1,32 @@
+# NumberFormatting 
+
+## Assignment
+Format a number for a specific locale. For example, the number 12345.124 should be formatted as `12,345.12` in the USA locale, but `12.345,78` in the German locale.
+
+## Solution
+```java
+DecimalFormat currencyFormatter = (DecimalFormat) NumberFormat.getInstance(Locale.getDefault());
+
+currencyFormatter.setMaximumFractionDigits(2);
+currencyFormatter.setMinimumFractionDigits(2);
+
+System.out.println(currencyFormatter.format(123456.12));
+```
+
+Depending on your default Locale the currencyFormatter will format the number in the right way for you.
+
+##### Will output:
```

Comment:
Using `Locale.getDefault()`allows you to get the System default, so it depends on your location. You can test it for other Locale's by using fixed locale values like: `Locale.GERMAN`or `Locale.US`for USA ;) You also can find the Local according to the language/country by using `Locale(String language)` or `Locale(String language, String country)` but this can also be found in [Documentation for Locale].
I will add this information so nobody get troubles with that.
[Documentation for Locale]:http://docs.oracle.com/javase/7/docs/api/java/util/Locale.html